nestle.py

```diff
@@ -560,13 +591,51 @@ class Sampler:
     given some existing set of points."""
 
     def __init__(self, loglikelihood, prior_transform, points, rstate,
-                 options):
+                 options, n_threads, pool):
         self.loglikelihood = loglikelihood
         self.prior_transform = prior_transform
         self.points = points
         self.rstate = rstate
         self.set_options(options)
-
+        self.n_threads = n_threads
+        self.pool = pool
+        self.queue = []
+        self.submitted = 0
+        self.cancelled = 0
+        self.unused = 0
+        self.used = 0
+
+    def empty_queue(self):
+        """Dump all operations on the queue."""
+
+        while self.queue:
+            x, v, f = self.queue.pop()
+            if f.cancel():
+                self.cancelled += 1
+            else:
+                self.unused += 1
+
+    def fill_queue(self):
+        """Fill up the queue with operations."""
+
+        while len(self.queue)<self.n_threads:
+            x = self.propose_point()
+            v = self.prior_transform(x)
+            self.queue.append((x, v, self.pool.submit(self.loglikelihood, v)))
+            self.submitted += 1
+
+    def get_point_value(self, should_update):
+        """ Get evaluation sequentially from the queue. If we should
+            update our proposal distribution, do not refill the queue."""
+
+        if not self.queue:
+            self.fill_queue()
+        x, v, f = self.queue.pop(0)
+        r = f.result()
```

Comment:
In terms of statistics, this change shouldn't affect the result in any way. The update schedule for the ellipsoids is purely an algorithmic choice. The number of iterations between ellipsoids is often (as in the default implementation) taken to be fixed, but it could easily be variable, set by some density criteria, etc.